mining.py

```diff
@@ -84,7 +84,7 @@ def target_to_hex(target):
 
 IDEAL_BLOCK_TIME = 10 * 60
 
-State = namedtuple('State', 'height wall_time timestamp bits chainwork fx '
+State = namedtuple('State', 'height wall_time timestamp max_timestamp bits chainwork fx '
```

Comment:
Can we avoid adding another member?

Reply:
Well, I could probably do something hackier like `max(state.timestamp for state in states[-100:-1])`.  Anyway I think @zawy12 may be convincing me that the max timestamp approach isn't needed...  So I'm OK making that change if you prefer?

Reply:
I'd prefer that.  If every algo started putting stuff in the state it would become problematic.

Reply:
Done!